### PR TITLE
Borg movement now has sounds

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
adds sound effects to cyborgs when they move
Demo: 
https://imgur.com/a/7Mu98pQ

Open to feedback on this sound so please tell me what you think.
## Why It's Good For The Game
a hunk of metal moving at you without any sound is frightening
## Changelog
:cl: grungussuss
sound: cyborgs now have sounds to indicate they are moving
/:cl:

### CODE REVIEW

**Comment by Rhials** [comment]
You need to add these sounds and their attributions to the attributions.txt file in tgstation/sound. Include what license they're under alongside a link to the freesound pages.

Also, could you record a video of this in action? I'd love to hear what this sounds like and the website links aren't giving me the full picture.

**Comment by Sadboysuss** [comment]
> You need to add these sounds and their attributions to the attributions.txt file in tgstation/sound. Include what license they're under alongside a link to the freesound pages.
> 
> Also, could you record a video of this in action? I'd love to hear what this sounds like and the website links aren't giving me the full picture.

yep, I'm just trying to figure out how to post it since I haven't compiled it yet and checked ingame, that's why it's marked as draft right now

**Comment by moocowswag** [comment]
This does not sound like it would be very pleasant to hear if multiple borgs were walking around in the same area.

**Comment by san7890** [comment]
maybe volume down a bit then? unsure. i think it's okay the way it is in the video but i can see how stacking makes it sound a bit worse

**Comment by moocowswag** [comment]
https://github.com/tgstation/tgstation/assets/62126254/03f3dfd7-f748-4ebc-b0c5-c061ef37b771

i slowed your video down to 0.6x speed and this is what it sounds like, experiment.

**Comment by Sadboysuss** [comment]
> point6timesspeed.mp4 
> i slowed your video down to 0.6x speed which also lowered the pitch and it sounded alittle less distracting, maybe also make it so the noise isnt played so constant/repeats so fast.
> 
> Btw you dont need to post your video on some third party website you can just drag and drop the file into whatever text box you are working on in github and the website will do the rest.

My file was too big so I used imgur

I updated the video to reflect the new change

**Comment by zxaber** [comment]
You should add a variable to the borg mob that carries the path to the footstep sound, and another on each model that carries an override. When the borg model is applied, copy the override var onto the borg's own footstep sound. If a borg gets model reset, set their footstep sound to the initial value.

This will make it so that adding sounds later is cleaner. Engineer borgs, for example, should have a metalic spider walking sound effect but someone can add that at a later date.

**Comment by Sadboysuss** [comment]
> You should add a variable to the borg mob that carries the path to the footstep sound, and another on each model that carries an override. When the borg model is applied, copy the override var onto the borg's own footstep sound. If a borg gets model reset, set their footstep sound to the initial value.
> 
> This will make it so that adding sounds later is cleaner. Engineer borgs, for example, should have a metalic spider walking sound effect but someone can add that at a later date.

it's a great idea but I wanted to do something really simple and non-complicated, so respectfully, until a maintainer tells me to do it I'll keep it as it is.

**Comment by Zergspower** [comment]
May I suggest making this a module trait, as in it's defined on the borg skin itself?

Downstream we have of course we've dozens of borg skins and based on the type of skin used they have specific traits added to them. e.g. sitting, special hat placement, custom 'broken' sprites etc.

image

Would allow for different sounds based on the borg type as well, e.g. treaded borgs being smol tanks and the like

**Comment by san7890** [comment]
the whole fucking way `/datum/element/footstep` works needs a deep refactor and this PR isn't really appropriate to be the one to do it - needs it's own dedicated effort. sadly it means we'll tacking onto a really poor framework but it's not reasonable to expect every nice sound to have a big concerted code effort behind it

File: code/__DEFINES/footsteps.dm [hunk 1]
```diff
@@ -28,6 +28,8 @@
 #define FOOTSTEP_MOB_RUST "footstep_rust"
 #define FOOTSTEP_OBJ_MACHINE "footstep_machine"
 #define FOOTSTEP_OBJ_ROBOT "footstep_robot"
+#define FOOTSTEP_OBJ_CYBORG_FLY "footstep_cyborg_fly"
```
Comment (san7890) [inline code/__DEFINES/footsteps.dm:31]: i think you forgot to stage/push the code where you actually add the `CYBORG_FLY` sounds. they're in your video but there doesn't appear to be any code for it in here.

Comment (Sadboysuss) [reply code/__DEFINES/footsteps.dm:31]: I wanted to add it at first but I decided on keeping it really simple for now, I'll remove the define.

**Comment by carshalash** [comment]
It's obnoxiously loud and unpleasant.

**Comment by OdoSendai** [comment]
> > It's obnoxiously loud and unpleasant.
> 
> any suggestions on a fix?

Revert change, or switch to the normal spaceman "footsteps" walking sound until a suitable replacement is found.

I get the impression the person who made this PR doesn't pay borg, because it is insane to me that someone thought forcing someone to listen to this sound non-stop for an hour+ is reasonable.  The sound is so annoying I don't think it is unreasonable to describe the game as unplayable as borg until it is changed.

**Comment by carshalash** [comment]
The loop is far too frequent and way too loud. The audio is far louder than shown in your preview.

**Comment by DaydreamIQ** [comment]
+1 To the sound issues, when this dropped just hearing a borg moving around was giving me a migrane something fierce. On account of how loud it is it loops just so badly that you can tell when the soundbyte cuts off

**Pull request merged**